gtest_parallel_tests.py

```diff
@@ -446,5 +446,207 @@ def root():
+    self.assertIn("--gtest_also_run_disabled_tests", mock_Popen.last_invocation)
+
+  def test_runs_only_failed_tests_when_asked(self):
+    fake_tests = {
+        "fake_unittests": {
+            "FakeTest": {
+                "FailedTest": None,
+                "Test": 1,
+            }
+        }
+    }
+    mock_Popen = self.MockPopen(fake_tests)
+
+    with guard_patch_module('subprocess.Popen', mock_Popen):
+      tasks = gtest_parallel.find_tests(
+          fake_tests.keys(), [],
+          self.MockOptions(failed=False),
+          self.MockTestTimes(fake_tests)
+      )
+    self.assertEqual(len(tasks), 2)
+    self.assertEqual(sorted([task.test_name for task in tasks]),
+                     ["FakeTest.FailedTest", "FakeTest.Test"])
+    self.assertEqual(sorted([task.last_execution_time for task in tasks]),
+                     [None, 1])
+
+    with guard_patch_module('subprocess.Popen', mock_Popen):
+      tasks = gtest_parallel.find_tests(
+          fake_tests.keys(), [],
+          self.MockOptions(failed=True),
+          self.MockTestTimes(fake_tests)
+      )
+    self.assertEqual(len(tasks), 1)
+    self.assertEqual(tasks[0].test_binary, "fake_unittests")
+    self.assertEqual(tasks[0].test_name, "FakeTest.FailedTest")
+    self.assertIsNone(tasks[0].last_execution_time)
+
+  def test_processes_gtest_filter(self):
+    fake_tests = { "fake_unittests": {} }
+    mock_Popen = self.MockPopen(fake_tests)
+
+    with guard_patch_module('subprocess.Popen', mock_Popen):
+      gtest_parallel.find_tests(
+          fake_tests.keys(), [],
+          self.MockOptions(gtest_filter='SomeFilter'),
+          self.MockTestTimes(fake_tests)
+      )
+    self.assertIn('--gtest_filter=SomeFilter', mock_Popen.last_invocation)
+
+    with guard_patch_module('subprocess.Popen', mock_Popen):
+      gtest_parallel.find_tests(
+          fake_tests.keys(), [],
+          self.MockOptions(gtest_filter=""),
+          self.MockTestTimes(fake_tests)
+      )
+    self.assertNotIn('--gtest_filter=SomeFilter', mock_Popen.last_invocation)
+
+  def test_processes_gtest_color(self):
+    fake_tests = {
+        "fake_unittests": {
+            "FakeTest": {
+                "TestSomething": None,
+            }
+        }
+    }
+    mock_Popen = self.MockPopen(fake_tests)
+    with guard_patch_module('subprocess.Popen', mock_Popen):
+      tasks = gtest_parallel.find_tests(
+          fake_tests.keys(), [],
+          self.MockOptions(gtest_color='Lemur'),
+          self.MockTestTimes(fake_tests)
+      )
+    self.assertEqual(len(tasks), 1)
+    self.assertIn('--gtest_color=Lemur', tasks[0].test_command)
+
+  def test_processes_repeat(self):
+    fake_tests = {
+        "fake_unittests": {
+            "FakeTest": {
+                "TestSomething": None,
+            }
+        }
+    }
+    mock_Popen = self.MockPopen(fake_tests)
+
+    with guard_patch_module('subprocess.Popen', mock_Popen):
```

Comment:
There's a lot of code repetition here (I've definitely seen this block more than once). Can you have a `def mock_find_tests(mock_popen, fake_tests, options):` method and remove all corresponding `with` blocks with one-liners?